File: facultate/an II/af/examen/parcurgeri/Parcurgeri/mod_liste_adiacenta.cpp
#include<iostream>
#include<fstream>
#include<vector>
using namespace std;

void citire(int &n,vector<int> *&la,int orientat=0,const char *nume_fisier="graf.in"){
	int i,j,x,y,m;
	ifstream f(nume_fisier);
	f>>n>>m;
    la=new vector<int>[n];
 
	while(f>>x>>y){
		x--;y--;
		la[x].push_back(y);
		if (not orientat)
		    la[y].push_back(x);
	}
	f.close();
}
void afisare_liste(int n, vector<int> *la){
	int i,j;
	for(i=0;i<n;i++){
		cout<<i+1<<": ";
		for(j=0;j<la[i].size();j++)
			cout<<la[i][j]+1<<" ";
		cout<<endl;
	}
}

File: facultate/an II/af/examen/parcurgeri/Parcurgeri/mod_liste_adiacenta.py
def afisare_liste(la):
    for (i,li) in enumerate(la):
        print("{}:".format(i+1), end=' ')
        for x in li:
            print('{}'.format(x+1), end= ' ')
     
        print()    

def citire(orientat=False,nume_fisier="graf.in"):
    n=0
    a=[]
    with open(nume_fisier) as f:
        linie = f.readline()
        n,m=(int(z) for z in linie.split())
        
        la=[[] for i in range(n)]#la=n*[[]] #!!!NU e ok, sunt aceeasi lista
   
        for linie in f:
            x,y=(int(z) for z in linie.split())
            x-=1; y-=1
            la[x].append(y)
            if not orientat:
                la[y].append(x)
         
        
    return n,la




